apps/somnia/src/commands/archive/char.ts

```diff
@@ -0,0 +1,140 @@
+import type {
+  ElementKey,
+  LocationCharacterKey,
+} from '@genshin-optimizer/gi/consts'
+import { allStats } from '@genshin-optimizer/gi/stats'
+import type { Interaction, MessageActionRowComponentBuilder } from 'discord.js'
+import {
+  ActionRowBuilder,
+  EmbedBuilder,
+  StringSelectMenuBuilder,
+  StringSelectMenuOptionBuilder,
+} from 'discord.js'
+import { clean, colors, talentlist } from '../archive'
+
+export function chararchive(
+  interaction: Interaction,
+  id: LocationCharacterKey,
+  name: string,
+  data: any,
+  args: string
+) {
+  //get element
+  let element = 'none'
+  const travelerelement = id.match(/Traveler(.*)F/)
+  if (!travelerelement) element = allStats.char.data[id].ele!
+  else element = travelerelement[1].toLowerCase()
+  //setup embed
+  const embed = new EmbedBuilder()
+    .setFooter({
+      text: 'Character Archive',
+    })
+    .setAuthor({
+      name: name,
+      iconURL: colors.element[element as ElementKey | 'none'].img,
+    })
+    .setColor(colors.element[element as ElementKey | 'none'].color)
+  //set contents
+  let text = ''
+  const talent = args
+
+  //character profile
+  if (talent === 'p') {
+    text =
```

Comment:
For each of these `if (talent ==...)` blocks, extract the body of the if-statement to a function like `text = passiveText(); embed= passiveEmbed()` or something